contracts/token/ERC721/ERC721.sol

```diff
@@ -174,8 +172,8 @@ contract ERC721 is ERC165, IERC721 {
      * @param _data bytes data to send along with a safe transfer check
      */
     function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data) public {
-        transferFrom(from, to, tokenId);
-        require(_checkOnERC721Received(from, to, tokenId, _data), "ERC721: transfer to non ERC721Receiver implementer");
+        require(_isApprovedOrOwner(msg.sender, tokenId));
```

Comment:
We should add a revert reason here. I think you can just copy the one in `transferFrom`, right @nventuro?